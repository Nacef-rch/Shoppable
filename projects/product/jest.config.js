// eslint-disable-next-line @typescript-eslint/no-var-requires
const baseConfig = require('../../jest.config');

module.exports = {
    ...baseConfig,
    rootDir: '../..',

    roots: ['<rootDir>/projects/product'],
    collectCoverageFrom: [
        '<rootDir>/projects/product/**/*.ts',
        '!**/*.module.ts',
        '!**/*.stories.ts'
    ]
};
