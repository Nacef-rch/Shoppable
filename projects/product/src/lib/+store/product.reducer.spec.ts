import * as productAction from '@product/+store/product.actions';
import { productInitialState, productReducer as reducer } from '@product/+store/product.reducer';
import * as productMock from '@product/mocks/products.mock.json';

describe('productReducer', () => {
    describe('undefined action', () => {
        it('should return the default state', () => {
            // GIVEN
            // WHEN
            const result = reducer(undefined, {} as any);
            // THEN
            expect(result).toMatchSnapshot();
        });

        describe('PRODUCT IMPORT SUCCESS , START AND FAIL', () => {
            it('`IMPORT_SUCCESS`', () => {
                // GIVEN
                const action = productAction.IMPORT_SUCCESS({
                    successMessage: 'success'
                });
                // WHEN
                const result = reducer(productInitialState, action);
                // THEN
                expect(result).toMatchSnapshot();
            });
        });
        it('`IMPORT_FAIL`', () => {
            // GIVEN
            const errorMessage = 'Error';
            const action = productAction.IMPORT_FAIL({ errorMessage });
            // WHEN
            const result = reducer(productInitialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
        it('IMPORT_START', () => {
            // GIVEN
            const action = productAction.IMPORT_START(productMock.importMock);
            // WHEN
            const result = reducer(productInitialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });
    describe('DELETE_PRODUCT', () => {
        it('DELETE_PRODUCT', () => {
            // GIVEN
            const action = productAction.DELETE_PRODUCT(productMock.deleteMock);
            // WHEN
            const result = reducer(productInitialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });
    describe('FETECH STORE SUCESS & START', () => {
        it('FETCH_STORE_PRODUCTS_SUCCESS', () => {
            // GIVEN
            const action = productAction.FETCH_STORE_PRODUCTS_SUCCESS(
                productMock.storeProductsMock
            );
            // WHEN
            const result = reducer(productInitialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
        it('FETCH_STORE_PRODUCTS_START', () => {
            // GIVEN
            const action = productAction.FETCH_STORE_PRODUCTS_START();
            // WHEN
            const result = reducer(productInitialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });
    describe('FETECH CATEGORY SUCESS & START', () => {
        it('FETCH_STORE_CATEGORY_SUCCESS', () => {
            // GIVEN
            const action = productAction.FETCH_STORE_CATEGORY_SUCCESS(productMock.storeCategories);
            // WHEN
            const result = reducer(productInitialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
        it('FETCH_STORE_CATEGORY_START', () => {
            // GIVEN
            const action = productAction.FETCH_STORE_CATEGORY_START();
            // WHEN
            const result = reducer(productInitialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });
    describe('CHANGE_PRODUCT_STOCK', () => {
        it('CHANGE_PRODUCT_STOCK', () => {
            // GIVEN
            const action = productAction.CHANGE_PRODUCT_STOCK(productMock.QuantitySock);
            // WHEN
            const result = reducer(productInitialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });
    describe('LOADING', () => {
        it('LOADING_START', () => {
            // GIVEN
            const action = productAction.LOADING_START();
            // WHEN
            const result = reducer(productInitialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
        it('LOADING_STOP', () => {
            // GIVEN
            const action = productAction.LOADING_STOP();
            // WHEN
            const result = reducer(productInitialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });
    describe('ERROR+SUCESS', () => {
        it('CLEAR_ERROR', () => {
            // GIVEN
            const action = productAction.CLEAR_ERROR();
            // WHEN
            const result = reducer(productInitialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
        it('CLEAR_SUCCESS', () => {
            // GIVEN
            const action = productAction.CLEAR_SUCCESS();
            // WHEN
            const result = reducer(productInitialState, action);
            // THEN
            expect(result).toMatchSnapshot();
        });
    });
});
