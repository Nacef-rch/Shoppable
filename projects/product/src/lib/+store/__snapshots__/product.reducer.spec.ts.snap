// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productReducer CHANGE_PRODUCT_STOCK CHANGE_PRODUCT_STOCK 1`] = `
Object {
  "loading": false,
  "productError": null,
  "productSuccess": null,
  "storeCategories": Array [],
  "storeProducts": Array [],
}
`;

exports[`productReducer DELETE_PRODUCT DELETE_PRODUCT 1`] = `
Object {
  "loading": false,
  "productError": null,
  "productSuccess": null,
  "storeCategories": Array [],
  "storeProducts": Array [],
}
`;

exports[`productReducer ERROR+SUCESS CLEAR_ERROR 1`] = `
Object {
  "loading": false,
  "productError": null,
  "productSuccess": null,
  "storeCategories": Array [],
  "storeProducts": Array [],
}
`;

exports[`productReducer ERROR+SUCESS CLEAR_SUCCESS 1`] = `
Object {
  "loading": false,
  "productError": null,
  "productSuccess": null,
  "storeCategories": Array [],
  "storeProducts": Array [],
}
`;

exports[`productReducer FETECH CATEGORY SUCESS & START FETCH_STORE_CATEGORY_START 1`] = `
Object {
  "loading": true,
  "productError": null,
  "productSuccess": null,
  "storeCategories": Array [],
  "storeProducts": Array [],
}
`;

exports[`productReducer FETECH CATEGORY SUCESS & START FETCH_STORE_CATEGORY_SUCCESS 1`] = `
Object {
  "loading": false,
  "productError": null,
  "productSuccess": null,
  "storeCategories": Array [
    Object {
      "lostOfCategories": Array [
        "firs",
        "second",
      ],
      "storeId": "string",
    },
  ],
  "storeProducts": Array [],
}
`;

exports[`productReducer FETECH STORE SUCESS & START FETCH_STORE_PRODUCTS_START 1`] = `
Object {
  "loading": true,
  "productError": null,
  "productSuccess": null,
  "storeCategories": Array [],
  "storeProducts": Array [],
}
`;

exports[`productReducer FETECH STORE SUCESS & START FETCH_STORE_PRODUCTS_SUCCESS 1`] = `
Object {
  "loading": false,
  "productError": null,
  "productSuccess": null,
  "storeCategories": Array [],
  "storeProducts": Array [
    Object {
      "categoryId": "string",
      "description": "string",
      "imageUrl": "string",
      "name": "string",
      "position": 1,
      "productId": "string",
      "quantityInStock": 27,
      "storeId": "string",
      "unitPrice": 26,
      "userHandle": "string",
    },
    Object {
      "categoryId": "string",
      "description": "string",
      "imageUrl": "string",
      "name": "string",
      "position": 2,
      "productId": "string",
      "quantityInStock": 27,
      "storeId": "string",
      "unitPrice": 26,
      "userHandle": "string",
    },
  ],
}
`;

exports[`productReducer LOADING LOADING_START 1`] = `
Object {
  "loading": true,
  "productError": null,
  "productSuccess": null,
  "storeCategories": Array [],
  "storeProducts": Array [],
}
`;

exports[`productReducer LOADING LOADING_STOP 1`] = `
Object {
  "loading": false,
  "productError": null,
  "productSuccess": null,
  "storeCategories": Array [],
  "storeProducts": Array [],
}
`;

exports[`productReducer undefined action \`IMPORT_FAIL\` 1`] = `
Object {
  "loading": false,
  "productError": "Error",
  "productSuccess": null,
  "storeCategories": Array [],
  "storeProducts": Array [],
}
`;

exports[`productReducer undefined action IMPORT_START 1`] = `
Object {
  "loading": true,
  "productError": null,
  "productSuccess": null,
  "storeCategories": Array [],
  "storeProducts": Array [],
}
`;

exports[`productReducer undefined action PRODUCT IMPORT SUCCESS , START AND FAIL \`IMPORT_SUCCESS\` 1`] = `
Object {
  "loading": false,
  "productError": null,
  "productSuccess": "success",
  "storeCategories": Array [],
  "storeProducts": Array [],
}
`;

exports[`productReducer undefined action should return the default state 1`] = `
Object {
  "loading": false,
  "productError": null,
  "productSuccess": null,
  "storeCategories": Array [],
  "storeProducts": Array [],
}
`;
